sklearn/pipeline.py

```diff
@@ -627,6 +627,10 @@ def _pairwise(self):
         # check if first estimator expects pairwise input
         return getattr(self.steps[0][1], '_pairwise', False)
 
+    @property
+    def n_features_in_(self):
+        return self.steps[0][1].n_features_in_
```

Comment:
check is fitted?

Reply:
`self.steps[0][1]` corresponds to the first estimator, which will handle the `check_is_fitted` call there.

We don't have a uniform way of checking whether a pipeline is fitted anyway

Reply:
uhm this is annyoing. I recall that we were counting on `Pipeline` to have `n_features_in_` attribute to be able to call `check_is_fitted(pipe)` on it.

Reply:
> we were counting on Pipeline to have n_features_in_ attribute to be able to call check_is_fitted(pipe) on it.

That's not a solid strategy IMO, because n_features_in_ is valid as long as the first step is fitted. The other steps don't matter for n_features_in_.

However, a pipeline is fitted when *all* the steps are fitted.

So you may very well have a non-fitted pipeline with a valid and reliable n_features_in_ attribute.

Reply:
I think that it was one of my concern by then.